src/main/java/com/mercateo/sqs/utils/message/handling/MessageWrapper.java

```diff
@@ -0,0 +1,52 @@
+package com.mercateo.sqs.utils.message.handling;
+
+import com.amazonaws.services.sqs.AmazonSQS;
+import com.amazonaws.services.sqs.model.ChangeMessageVisibilityRequest;
+import com.amazonaws.services.sqs.model.ChangeMessageVisibilityResult;
+
+import io.awspring.cloud.messaging.listener.Acknowledgment;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+
+import org.springframework.messaging.Message;
+
+@RequiredArgsConstructor
+public class MessageWrapper<I> {
+
+    @NonNull
+    @Getter
+    private final Message<I> message;
+
+    private final AtomicBoolean acknowledged = new AtomicBoolean(false);
```

Comment:
What is the advantage of heaving an atomic boolean instead of a native boolean? The manipulating methods are already synchronized.

Reply:
replaced with boolean https://github.com/Mercateo/sqs-utils/pull/23/commits/d1b1b9b55c530660e6a0205e37378b845ef7bfe5